packages/protocol/test/unit/utils/StreamPartitionID.test.ts

```diff
@@ -0,0 +1,188 @@
+import SPID  from '../../../src/utils/StreamPartitionID'
+
+describe('StreamPartitionID', () => {
+    const STREAM_ID = 'test-stream-id'
+    const PARTITION = 1
+
+    it('can use object destructuring', () => {
+        const spid = new SPID(STREAM_ID, PARTITION)
+        const { id, partition, key } = spid
+        expect(id).toBe(STREAM_ID)
+        expect(partition).toBe(PARTITION)
+        expect(key).toBe(spid.toString())
+    })
+
+    it('defaults partition to 0', () => {
+        const spid = new SPID(STREAM_ID)
+        expect(spid.id).toEqual(STREAM_ID)
+        expect(spid.partition).toBe(0)
+    })
+
+    it('lowercases streamId', () => {
+        const spid = new SPID(STREAM_ID.toUpperCase())
+        expect(spid.id).toEqual(STREAM_ID)
+        expect(spid.partition).toBe(0)
+    })
+
+    it('can use array destructuring with split', () => {
```

Comment:
Thinking out loud here: if we can already do object destructuring, i.e.
```ts
const { id, partition} = spid
```
why do we need array destructing and thus method `split`?

Reply:
array destructuring is a bit cleaner if you don't want to use the defined `id`, `partition` names. Paritition is probably fine but `id` might already exist.
```js
const { id: streamId, partition: streamPartition } = spid
// vs
const [streamId, streamPartition] = spid.split()
```

Reply:
Aaah good point. :+1: